Spatial_Engine/Query/CurveLayout.cs

```diff
@@ -0,0 +1,85 @@
+﻿/*
+ * This file is part of the Buildings and Habitats object Model (BHoM)
+ * Copyright (c) 2015 - 2020, the respective contributors. All rights reserved.
+ *
+ * Each contributor holds copyright over their respective contributions.
+ * The project versioning (Git) records all such contribution source information.
+ *                                           
+ *                                                                              
+ * The BHoM is free software: you can redistribute it and/or modify         
+ * it under the terms of the GNU Lesser General Public License as published by  
+ * the Free Software Foundation, either version 3.0 of the License, or          
+ * (at your option) any later version.                                          
+ *                                                                              
+ * The BHoM is distributed in the hope that it will be useful,              
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of               
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the                 
+ * GNU Lesser General Public License for more details.                          
+ *                                                                            
+ * You should have received a copy of the GNU Lesser General Public License     
+ * along with this code. If not, see <https://www.gnu.org/licenses/lgpl-3.0.html>.      
+ */
+
+using System;
+using System.Linq;
+using System.Collections.Generic;
+using System.ComponentModel;
+using BH.oM.Reflection.Attributes;
+using BH.oM.Base;
+using BH.oM.Geometry;
+using BH.oM.Spatial.Layouts;
+using BH.Engine.Geometry;
+
+namespace BH.Engine.Spatial
+{
+    public static partial class Query
+    {
+        /***************************************************/
+        /**** Public Methods                            ****/
+        /***************************************************/
+
+        [Description("Queries the curves from the Layout based on the layout and a set of region curves and opening curves associated with it.")]
+        [Input("explicitCurveLayout", "The layout object to query the curves from.")]
+        [Input("hostRegionCurves", "The region curves of the objects associated with the layout. Unused for Explicit layouts.")]
+        [Input("openingCurves", "Optional opening curves in the region. Unused for ExplicitLayout.")]
+        [Output("curves", "The Curves stored in the ExplicitLayout object.")]
+        public static List<ICurve> CurveLayout(this OffsetCurveLayout offsetCurveLayout, IEnumerable<ICurve> hostRegionCurves, IEnumerable<ICurve> openingCurves = null)
```

Comment:
As discussed, lets add a minimum offset parameter to all those methods, default to 0. Then use the maximum of the provided minimum and the value on the layout